flows/definition/flow.go

```diff
@@ -102,68 +101,72 @@ func (f *flow) validate() error {
 
 // Inspect enumerates dependencies, checks that they exist
 func (f *flow) Inspect(sa flows.SessionAssets) error {
-	return f.doInspect(sa, false, nil)
+	return f.doInspect(sa, false, nil, nil)
 }
 
 // InspectRecursively checks that all of this flow's dependencies exist, and all our flow dependencies are also valid
 func (f *flow) InspectRecursively(sa flows.SessionAssets, missing func(assets.Reference)) error {
-	return f.doInspect(sa, true, missing)
+	seen := make(map[assets.FlowUUID]bool)
+
+	return f.doInspect(sa, true, seen, missing)
 }
 
 type brokenDependency struct {
 	dependency assets.Reference
 	reason     error
 }
 
-func (f *flow) doInspect(sa flows.SessionAssets, recursive bool, missing func(assets.Reference)) error {
-	// if this flow has already been inspected, don't need to do it again - avoid unnecessary work
-	// but also prevents looping if recursively inspecting flows
-	if f.inspected {
```

Comment:
need to be able to `Inspect` the same flow more than once - with and without assets